src/pages/rss.ts

```diff
@@ -0,0 +1,37 @@
+import { extname } from 'node:path'
+import { getPosts } from '@/lib/content'
+import rss from '@astrojs/rss'
+import type { APIRoute } from 'astro'
+
+export const GET: APIRoute = async ({ site, url }) => {
+  const posts = await getPosts()
+
+  return rss({
+    title: 'Engineering with Yagiz',
+    description:
+      "Here's a collection of posts about my thoughts, stories, ideas and experiences as a human, and an engineer working with different technologies.",
+    site: site as URL,
+    customData: `
+<lastBuildDate>${new Date().toUTCString()}</lastBuildDate>
+<docs>https://validator.w3.org/feed/docs/rss2.html</docs>
+<generator>Astro</generator>
+<language>en-us</language>
+<copyright>Yagiz Nizipli - yagiz.co</copyright>
+`,
+    items: posts.map((post) => ({
+      title: post.data.title,
+      customData: `
+<guid>${`/${post.slug}`}</guid>
+`,
+      link: `/${post.slug}`,
+      description: post.data.description,
+      pubDate: post.data.date,
+      enclosure: {
+        // TODO: can't set length to 0
```

Comment:
I have no idea how to handle that. In the nextjs version it was set to `0` although it seems just as incorrect because it should be the image size in bytes. I will crosspost on Astro's discord